docs/ref/tasks.txt

```diff
@@ -0,0 +1,334 @@
+=====
+Tasks
+=====
+
+.. versionadded:: 6.0
+
+.. module:: django.tasks
+   :synopsis: Django's built-in background task system.
+
+Task definition
+===============
+
+.. function:: task
+
+A decorator that defines a :class:`Task`::
+
+    from django.core.mail import send_mail
+    from django.tasks import task
+
+
+    @task
+    def email_users(emails, subject, message) -> int:
+        return send_mail(
+            subject=subject, message=message, from_email=None, recipient_list=emails
+        )
+
+In the above example, ``email_users`` is a :class:`Task` instance.
+
+``task`` accepts a number of arguments, which are used as-is for the ``Task``:
+
+* :attr:`priority <Task.priority>`
+* :attr:`backend <Task.backend>`
+* :attr:`queue_name <Task.queue_name>`
+* :attr:`enqueue_on_commit <Task.enqueue_on_commit>`
+
+If the ``Task`` is not valid, according to the backend,
+:exc:`django.tasks.exceptions.InvalidTaskError` is raised.
+
+.. class:: Task
+
+Represents a task to be run in the background. Tasks should be defined using
+the :func:`task` decorator.
+
+Attributes of ``Task`` cannot be modified.
+
+.. attribute:: Task.priority
+
+    The priority of the ``Task``. Priorities must be between -100 and 100,
+    where larger numbers are higher priority, and will be run sooner. Defaults
+    to 0.
+
+.. attribute:: Task.backend
+
+    The alias of the backend the ``Task`` should be enqueued to. Defaults
+    to ``"default"``.
+
+    This must match a backend defined in :setting:`BACKEND <TASKS-BACKEND>`.
+
+.. attribute:: Task.queue_name
+
+    The name of the queue the ``Task`` will be enqueued on to. Defaults to
+    ``"default"``.
+
+    This must match a queue defined in :setting:`QUEUES <TASKS-QUEUES>`.
+
+.. attribute:: Task.run_after
+
+    The earliest time the ``Task`` will be executed, or ``None`` to have no time set.
+
+    Defaults to ``None``.
+
+.. attribute:: Task.enqueue_on_commit
+
+    Whether the ``Task`` should be enqueued when the :ref:`<transaction <task_transactions>`
+    commits successfully, or immediately.
+
+    By default, the behavior of the backend's
+    :setting:`ENQUEUE_ON_COMMIT <TASKS-ENQUEUE_ON_COMMIT>` is used.
+
+.. attribute:: Task.name
+
+    An identifier for a ``Task``. Potentially useful for debugging.
+
+    A Task's name is not necessarily unique.
+
+    By default, this is is the name of the function decorated with :func:`task`.
```

Comment:
"By default" is confusing, since there is no way to set or change it (correct me if I'm wrong).

BTW, I do think it would be nice to allow to set it (in `@task` and `using`). If you prefer not, then perhaps just remove it and document `module_path`?